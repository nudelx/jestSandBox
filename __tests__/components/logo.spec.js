import React from  'react'
import { shallow, mount } from 'enzyme'
import Logo from '../../src/componets/logo'
import toJson from 'enzyme-to-json';
import enzymeSerializer from 'enzyme-to-json/serializer';
import objectToJson from 'object-to-json';

// TODO functions snapshot functions
// TODO custom functions snapshot
// TODO component snapshot
// TODO mock


describe('full logo test', () => {
  it('should include div with logo class', () => {
    const component = shallow(<Logo />)
    expect(component.containsMatchingElement(
      <div className="sw-logo"></div>
    )).toEqual(true);
  })

  it('should have only one div ', () => {
    const component = shallow(<Logo />)
    expect(component.children().length).toBe(0)
  })

  it('shouldn\'t have more props', () => {
    const component = shallow(<Logo />)
    expect(component.props()).toEqual({"className": "sw-logo"})
  })

  it('should match the dom presentation snapshot', () => {
    expect.addSnapshotSerializer(enzymeSerializer)
    const component = mount(<Logo />)
    expect(component).toMatchSnapshot()
  })

  it('should match the dom presentation snapshot Î© ', () => {
    const component = mount(<Logo />)
    expect(toJson(component)).toMatchSnapshot()
  })
})
