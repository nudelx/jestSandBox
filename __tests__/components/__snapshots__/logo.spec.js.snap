// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full logo test should match the dom presentation snapshot 1`] = `
<Logo>
  <div
    className="sw-logo"
  />
</Logo>
`;

exports[`full logo test should match the dom presentation snapshot Î©  1`] = `
<Logo>
  <div
    className="sw-logo"
  />
</Logo>
`;
